smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/ReactiveKafkaProducer.java

```diff
@@ -87,6 +92,10 @@ public ReactiveKafkaProducer(Map<String, Object> kafkaConfiguration, String chan
 
         kafkaWorker = Executors.newSingleThreadExecutor(KafkaSendingThread::new);
         producer = new KafkaProducer<>(kafkaConfiguration, keySerializer, valueSerializer);
+        if (kafkaConfiguration.containsKey(ProducerConfig.TRANSACTIONAL_ID_CONFIG)) {
+            // TODO is this the right way ?
+            initTransactions().subscribeAsCompletionStage();
```

Comment:
No, it will trigger the operationimmediately, but won't wait for the completion. So, you may end up with something not totally initialized.

Reply:
I hesitated to run it on the starter thread. But I think it is better to run it on the sender thread and in case of failure report it to the channel.

Reply:
This should be fine, no? Since all sending actions are serialized on the sending thread, nothing can start actually sending before the init transactions process completes -- asuming that `Producer.initTransactions()` is synchronous, which I don't know :-)